### ISSUE TITLE
Frequent crashes of Sublime Text 3 when changing git branches and/or popping git stash.

### ISSUE THREAD

**Issue by nathanpeck** [titoBouzout/SideBarEnhancements#264]
I've noticed a frequent pattern of Sublime Text 3 hanging with a beachball forever after switching git branches or popping a git stash. This behavior stops if I disable SideBarEnhancements, and starts again when I reenable it.

Not sure whether the issues lies inside SideBarEnhancements or in ST3 itself.

**Comment by titoBouzout** [comment]
okei lets debug it

1. Uninstall this package 
2  download this package as is, to your Packages folder /Packages/SideBarEnhancements
3. Remove this class https://github.com/titoBouzout/SideBarEnhancements/blob/st3/SideBar.py#L53-L67 (select the class and just delete it.
4. Restart.. 
5. Still crashes?

(we will remove code till we find it)

**Comment by nathanpeck** [comment]
So I did as you suggested, removed those lines, and put it through a quick stress test of switching between feature branches, and doing git stash and git stash pop while having folders expanded on the sidebar and I was not able to reproduce the hang.

But then I added the lines back in, restarted, and still wasn't able to reproduce it so most likely I just haven't repeated the specific conditions that trigger the hang. I'll keep using the modified version for a bit and let you know if I repro it again.
